autometa/taxonomy/gtdb.py

```diff
@@ -0,0 +1,362 @@
+"""
+# License: GNU Affero General Public License v3 or later
+# A copy of GNU AGPL v3 should have been included in this software package in LICENSE.txt.
+
+File containing definition of the GTDB class and containing functions useful for handling GTDB taxonomy databases
+"""
+
+import gzip
+import logging
+import os
+import re
+import subprocess
+import shutil
+import tarfile
+
+from typing import Dict, List, Set, Tuple
+from itertools import chain
+from tqdm import tqdm
+
+import pandas as pd
+import multiprocessing as mp
+
+from autometa.common.utilities import file_length
+from autometa.common.external import diamond
+from autometa.taxonomy.database import TaxonomyDatabase
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_gz_file(filepath) -> bool:
+    """
+    Check if the given file is gzipped compressed or not.
+
+    Parameters
+    ----------
+    filepath : str
+        Filepath to check
+
+    Returns
+    -------
+    bool
+        True if file is gzipped else False
+    """
+    # https://stackoverflow.com/a/47080739
+    with open(filepath, "rb") as test_f:
+        return test_f.read(2) == b"\x1f\x8b"
+
+
+def create_gtdb_db(reps_faa: str, dbdir: str):
```

Comment:
✏️ :memo: This needs a docstring

Reply:
And a typehint

```suggestion
def create_gtdb_db(reps_faa: str, dbdir: str) -> str:
```

Reply:
It is not clear whether `reps_faa` is supposed to be a file path, directory path or tarball. This usage (and variable name) seems too ambiguous

Reply:
It can be a tarball or a directory path.